src/widget/wstemcontrol.cpp

```diff
@@ -0,0 +1,164 @@
+    }
+}
+} // namespace
+
+WStemControlBox::WStemControlBox(
+        const QString& group, QWidget* parent)
+        : WWidgetGroup(parent), m_group(group), m_hasStem(false), m_displayed(true) {
+    auto pLayout = new QVBoxLayout(this);
+    pLayout->setSpacing(0);
+    pLayout->setContentsMargins(0, 0, 0, 0);
+    setLayout(pLayout);
+
+    setObjectName("StemControlBox");
+
+    setWindowFlag(Qt::ToolTip, true);
+    setWindowFlag(Qt::FramelessWindowHint, true);
+
+    setAttribute(Qt::WA_OpaquePaintEvent);
+    setAttribute(Qt::WA_NoSystemBackground);
+    setAttribute(Qt::WA_TranslucentBackground);
+    setFocusPolicy(Qt::NoFocus);
+}
+
+void WStemControlBox::setDisplayed(bool displayed) {
+    m_displayed = displayed;
+    emit displayedChanged(m_displayed);
+}
+
+void WStemControlBox::slotTrackLoaded(TrackPointer track) {
+    m_hasStem = false;
+    if (!track) {
+        return;
+    }
+
+    auto stemInfo = track->getStemInfo();
+
+    if (stemInfo.isEmpty()) {
+        return;
+    }
+
+    m_hasStem = true;
+
+    int stemCount = qMin(m_stemControl.size(),
+            static_cast<std::size_t>(stemInfo.size()));
+    for (int stemIdx = 0; stemIdx < stemCount; stemIdx++) {
+        m_stemControl[stemIdx]->setStem(stemInfo[stemIdx]);
+    }
+}
+
+void WStemControlBox::addControl(QWidget* control) {
+    auto pWidget = std::make_unique<WStemControl>(control, this, m_group, m_stemControl.size());
+    layout()->addWidget(pWidget.get());
+    m_stemControl.push_back(std::move(pWidget));
+}
+
+WStemControl::WStemControl(QWidget* widgetGroup, QWidget* parent, const QString& group, int stemIdx)
+        : WWidget(parent),
+          m_widget(widgetGroup),
+          m_mutedCo(std::make_unique<ControlProxy>(
+                  group, QString("stem_%1_mute").arg(stemIdx + 1))) {
+    auto pLayout = new QHBoxLayout(this);
+    pLayout->setSpacing(0);
+    pLayout->setContentsMargins(0, 0, 0, 0);
+
+    m_widget->setParent(this);
+    setMinimumSize(m_widget->minimumSize());
+    setMaximumSize(m_widget->maximumSize());
+    setSizePolicy(m_widget->sizePolicy());
+
+    layout()->addWidget(m_widget);
+
+    m_mutedCo->connectValueChanged(this, [this](double value) {
+        QPalette pal = palette(); // we have to copy out the palette to edit it
+                                  // since it's const (probably for threadsafety)
+        QColor backgroundColor = pal.color(backgroundRole());
+        backgroundColor.setAlphaF(value == 1 ? 0.5 : 1.0);
+        pal.setColor(backgroundRole(), backgroundColor);
+        setAutoFillBackground(true);
+        setPalette(pal);
+    });
+}
+
+void WStemControl::setStem(const StemInfo& stemInfo) {
+    QPalette pal = palette(); // we have to copy out the palette to edit it
+                              // since it's const (probably for threadsafety)
+    QColor backgroundColor = stemInfo.getColor();
+    backgroundColor.setAlphaF(m_mutedCo->get() == 1 ? 0.5 : 1.0);
```

Comment:
```suggestion
    backgroundColor.setAlphaF(m_mutedCo->get() == 1 ? 0.5 : 1.0);
    pal.setColor(backgroundRole(), backgroundColor);
```
Without this, the stem controls appear white.

Reply:
Ah thanks, failed copy pasta :D